### ISSUE TITLE
Is ADT message supported

### ISSUE THREAD

**Issue by daya** [ruby-hl7/ruby-hl7#67]
Being a noob I can't figure out if ruby-hl7 supports ADT messages and its various segments https://datica.com/academy/hl7-201-the-admission-discharge-transfer-adt-message/

**Comment by mogox** [comment]
The list of segments it's pretty much this list: https://github.com/ruby-hl7/ruby-hl7/tree/master/lib/segments

I think it's a good idea to add it to the readme.

Do you need support for ADT messages? I think we could create a new gem similar to https://github.com/ruby-hl7/ruby-hl7-zps

**Comment by daya** [comment]
Yes ADT support is needed but isn't ruby-hl7-zps specific for ZPS segments or may be I don't understand how ruby-hl7 gem is architected as I did not find any message types also called **Data Definition Tables** 
[HL7-v26 Data Definition Tables Appendix A.pdf](https://github.com/ruby-hl7/ruby-hl7/files/974908/HL7-v26.Data.Definition.Tables.Appendix.A.pdf)

At a very high level do you have plan for ruby-hl7 to support profiles? Profile AFAIK is like a schema, where in units/hospitals define their extensions/variations/scopes to HL7v2 messages (may be in Z segments or elsewhere)

**Comment by mogox** [comment]
There is a gem that adds support to the segment `ZPS` and the gem is called `ruby-hl7-zps`. 
I'm not familiar with the `ADT` segment so I was wondering if you think we should include it as part of the base segments of if we should create a new gem so people who need support for `ADT` segments can include that gem also. 

What do you think?

Regarding profiles AFAIK, I haven't discuss anything with the rest of the team but I'm the one who is not working right now in health industry.
@ruby-hl7/contributors what do you think?

**Comment by mitch-lindsay** [comment]
So we don't have any native support for defining the structure for particular message types.  This library is geared to support creating well structured HL7 agnostic of any message patterns and it's up to the user to put the segments together in the right order for the particular message type.  I wouldn't necessarily be against defining a library of known message types though.  Typically I've generated ADT type messages using a pattern similar to this.
```
msg = HL7::Message.new

msg << MSH.builder(new_record? ? "ADT^A04" : "ADT^A08")
msg << EVN.builder
msg << PID.builder
msg << PV1.builder

msg.to_s
```

Regarding profiles, I'm curious if that hospital profile is proprietary and they wouldn't mind it being shared on the internet.  If so, that seems like the perfect fit for a new gem specific to the unit/hospital that imports all of their specific segments.
